File: src/de/epages/WebServices/PriceListAssignmentService/PriceListAssignmentServiceClient.java
package de.epages.WebServices.PriceListAssignmentService;

import de.epages.WebServices.PriceListAssignmentService.Stub.*;
import de.epages.WebServices.WebServiceConfiguration;

import java.util.logging.Logger;
import java.util.List;
import java.util.ArrayList;
import java.rmi.RemoteException;

public class PriceListAssignmentServiceClient extends WebServiceConfiguration  {
    private PriceListAssignmentService service;
    private Bind_PriceListAssignment_SOAPStub stub;
    private static Logger log = Logger.getLogger(PriceListAssignmentServiceClient.class.getName());

    /**
     * Class constructor
     */
    public PriceListAssignmentServiceClient() {
        service = new PriceListAssignmentServiceLocator();
        log.info("address specified by wsdl: " + service.getport_PriceListAssignmentAddress());
        log.info("using web service Url: " + WEBSERVICE_URL);

        try {
            stub = new Bind_PriceListAssignment_SOAPStub(new java.net.URL(WEBSERVICE_URL), service);

            // setting user-path and password of the shop
            stub.setUsername(WEBSERVICE_LOGIN);
            stub.setPassword(WEBSERVICE_PASSWORD);
        }
        catch (Exception e) {
            log.severe( e.toString() );
        }
    }

    /**
     * Gets QuantityDiscounts for a PriceList / Product combination.
     * If there is an error with a single PriceListAssignment, then this PriceListAssignment will be skipped and
     * it will be not part of there result list.
     *
     * @param productPaths      list of Products
     * @param pricelistPaths    list of PriceLists
     * @return              list of TGetQuantityDiscounts_Return
     * @throws              RemoteException
     * @see                 TGetQuantityDiscounts_Return
     */
    public List<TGetQuantityDiscounts_Return> getQuantityDiscounts(String[] productPaths, String[] pricelistPaths) throws RemoteException {
        log.info("getQuantityDiscounts called");
        TGetQuantityDiscounts_Return[] PriceListAssignments = stub.getQuantityDiscounts(productPaths, pricelistPaths);
        List result = new ArrayList();

        for (int i = 0; i < PriceListAssignments.length; i++) {
            TGetQuantityDiscounts_Return PriceListAssignment = PriceListAssignments[i];

            if (PriceListAssignment.getError() == null) {
                log.info("successfully retrieved QuantityDiscounts for Product: " + PriceListAssignment.getProduct());
                result.add(PriceListAssignment);
            } else {
                log.severe("an error occured (Epages Error):\n" + PriceListAssignment.getError().getMessage());
            }
        }
        return result;
    }

    /**
     * Creates QuantityDiscounts for specified Products and PriceLists.
     * If there is an error with a single PriceListAssignment, then this PriceListAssignment will be skipped and
     * it will be not part of there result list.
     *
     * @param           PriceListAssignments list of TSetQuantityDiscounts_Input
     * @return          list of TSetQuantityDiscounts_Return
     * @throws          RemoteException
     * @see             TSetQuantityDiscounts_Input
     * @see             TSetQuantityDiscounts_Return
     */
    public List<TSetQuantityDiscounts_Return> setQuantityDiscounts(List<TSetQuantityDiscounts_Input> PriceListAssignments) throws RemoteException {
        log.info("createPriceListAssignment called");
        TSetQuantityDiscounts_Input[] input = new TSetQuantityDiscounts_Input[PriceListAssignments.size()];
        for (int i = 0; i < PriceListAssignments.size(); i++) {
            input[i] =  PriceListAssignments.get(i);
        }

        TSetQuantityDiscounts_Return[] results = stub.setQuantityDiscounts(input);
        List<TSetQuantityDiscounts_Return> resultList = new ArrayList();

        for (int i = 0; i < results.length; i++) {
            TSetQuantityDiscounts_Return result = results[i];

            if (result.getError() == null) {
                log.info("successfully created QuantityDiscounts: " + result.getProduct() + "/" + result.getPriceList());
                resultList.add(result);
            } else {
                log.severe("an error occured (Epages Error):\n" + result.getError().getMessage());
            }
        }

        return resultList;
    }

    /**
     * Gets ValueDiscounts for a PriceList / Product combination.
     * If there is an error with a single PriceListAssignment, then this PriceListAssignment will be skipped and
     * it will be not part of there result list.
     *
     * @param productPaths      list of Products
     * @param pricelistPaths    list of PriceLists
     * @return              list of TGetQuantityDiscounts_Return
     * @throws              RemoteException
     * @see                 TGetValueDiscounts_Return
     */
    public List<TGetValueDiscounts_Return> getValueDiscounts(String[] productPaths, String[] pricelistPaths) throws RemoteException {
        log.info("getQuantityDiscounts called");
        TGetValueDiscounts_Return[] PriceListAssignments = stub.getValueDiscounts(productPaths, pricelistPaths);
        List result = new ArrayList();

        for (int i = 0; i < PriceListAssignments.length; i++) {
            TGetValueDiscounts_Return PriceListAssignment = PriceListAssignments[i];

            if (PriceListAssignment.getError() == null) {
                log.info("successfully retrieved ValueDiscounts for Product: " + PriceListAssignment.getProduct());
                result.add(PriceListAssignment);
            } else {
                log.severe("an error occured (Epages Error):\n" + PriceListAssignment.getError().getMessage());
            }
        }
        return result;
    }

    /**
     * Creates ValueDiscounts for specified Products and PriceLists.
     * If there is an error with a single PriceListAssignment, then this PriceListAssignment will be skipped and
     * it will be not part of there result list.
     *
     * @param           PriceListAssignments list of TSetValueDiscounts_Input
     * @return          list of TSetValueDiscounts_Return
     * @throws          RemoteException
     * @see             TSetValueDiscounts_Input
     * @see             TSetValueDiscounts_Return
     */
    public List<TSetValueDiscounts_Return> setValueDiscounts(List<TSetValueDiscounts_Input> PriceListAssignments) throws RemoteException {
        log.info("createPriceListAssignment called");
        TSetValueDiscounts_Input[] input = new TSetValueDiscounts_Input[PriceListAssignments.size()];
        for (int i = 0; i < PriceListAssignments.size(); i++) {
            input[i] =  PriceListAssignments.get(i);
        }

        TSetValueDiscounts_Return[] results = stub.setValueDiscounts(input);
        List<TSetValueDiscounts_Return> resultList = new ArrayList();

        for (int i = 0; i < results.length; i++) {
            TSetValueDiscounts_Return result = results[i];

            if (result.getError() == null) {
                log.info("successfully created ValueDiscounts: " + result.getProduct() + "/" + result.getPriceList());
                resultList.add(result);
            } else {
                log.severe("an error occured (Epages Error):\n" + result.getError().getMessage());
            }
        }

        return resultList;
    }

    /**
     * Gets ValueDiscounts for a PriceList / Product combination.
     * If there is an error with a single PriceListAssignment, then this PriceListAssignment will be skipped and
     * it will be not part of there result list.
     *
     * @param productPaths      list of Products
     * @param pricelistPaths    list of PriceLists
     * @return              list of TGetQuantityDiscounts_Return
     * @throws              RemoteException
     * @see                 TGetValueDiscounts_Return
     */
    public List<TGetScalePrices_Return> getScalePrices(String[] productPaths, String[] pricelistPaths) throws RemoteException {
        log.info("getQuantityDiscounts called");
        TGetScalePrices_Return[] PriceListAssignments = stub.getScalePrices(productPaths, pricelistPaths);
        List result = new ArrayList();

        for (int i = 0; i < PriceListAssignments.length; i++) {
            TGetScalePrices_Return PriceListAssignment = PriceListAssignments[i];

            if (PriceListAssignment.getError() == null) {
                log.info("successfully retrieved ScalePrices for Product: " + PriceListAssignment.getProduct());
                result.add(PriceListAssignment);
            } else {
                log.severe("an error occured (Epages Error):\n" + PriceListAssignment.getError().getMessage());
            }
        }
        return result;
    }

    /**
     * Creates ValueDiscounts for specified Products and PriceLists.
     * If there is an error with a single PriceListAssignment, then this PriceListAssignment will be skipped and
     * it will be not part of there result list.
     *
     * @param           PriceListAssignments list of TSetValueDiscounts_Input
     * @return          list of TSetValueDiscounts_Return
     * @throws          RemoteException
     * @see             TSetValueDiscounts_Input
     * @see             TSetValueDiscounts_Return
     */
    public List<TSetScalePrices_Return> setScalePrices(List<TSetScalePrices_Input> PriceListAssignments) throws RemoteException {
        log.info("createPriceListAssignment called");
        TSetScalePrices_Input[] input = new TSetScalePrices_Input[PriceListAssignments.size()];
        for (int i = 0; i < PriceListAssignments.size(); i++) {
            input[i] =  PriceListAssignments.get(i);
        }

        TSetScalePrices_Return[] results = stub.setScalePrices(input);
        List<TSetScalePrices_Return> resultList = new ArrayList();

        for (int i = 0; i < results.length; i++) {
            TSetScalePrices_Return result = results[i];

            if (result.getError() == null) {
                log.info("successfully created ScalePrices: " + result.getProduct() + "/" + result.getPriceList());
                resultList.add(result);
            } else {
                log.severe("an error occured (Epages Error):\n" + result.getError().getMessage());
            }
        }

        return resultList;
    }
}

File: dotnet/shop/EpagesWebServices/PriceListAssignmentServiceClient.cs
using System;
using EpagesWebServices.PriceListAssignmentServiceRef;
using System.Collections;

namespace EpagesWebServices
{
	/// <summary>
	/// Client that uses generated stub to call PriceListAssignment web services methods
	/// </summary>
	public class PriceListAssignmentServiceClient : WebServiceConfiguration
	{
		PriceListAssignmentService stub;

		public PriceListAssignmentServiceClient()
		{
			stub = new PriceListAssignmentService();
			stub.Credentials = new System.Net.NetworkCredential(WEBSERVICE_LOGIN, WEBSERVICE_PASSWORD);
			stub.PreAuthenticate = true;
			stub.Url = WEBSERVICE_URL;
		}

		/// <summary>
		/// use stub to create a QuantityDiscount via web services
		/// </summary>
		/// <param name="PriceListAssignments">Array of TSetQuantityDiscounts_Input</param>
		/// <returns>ArrayList of TSetQuantityDiscounts_Return</returns>
		public ArrayList setQuantityDiscounts(TSetQuantityDiscounts_Input[] PriceListAssignments) 
		{
			TSetQuantityDiscounts_Return[] PriceListAssignments_out = stub.setQuantityDiscounts(PriceListAssignments);

			ArrayList result = new ArrayList();

			for(int i = 0; i < PriceListAssignments_out.Length; i++)
			{
				TSetQuantityDiscounts_Return PriceListAssignment_out = PriceListAssignments_out[i];

				if (PriceListAssignment_out.Error == null) 
				{
					Console.WriteLine("successfully created QuantityDiscount for Product: " + PriceListAssignment_out.Product);
					result.Add(PriceListAssignment_out);
				} 
				else 
				{
					Console.WriteLine("an error occured (Epages Error):\n" + PriceListAssignment_out.Error.Message);
				}
			}

			return result;
		}

		/// <summary>
		/// use stub to get info of a QuantityDiscount via web services
		/// </summary>
		/// <param name="products">Array of product paths</param>
		/// <param name="pricelists">Array of pricelist paths</param>
		/// <returns>ArrayList of TGetQuantityDiscounts_Return</returns>
		public ArrayList getQuantityDiscounts(string[] products, string[] pricelists) 
		{
			TGetQuantityDiscounts_Return[] PriceListAssignments_out = stub.getQuantityDiscounts(products, pricelists);

			ArrayList result = new ArrayList();

			for(int i = 0; i < PriceListAssignments_out.Length; i++)
			{
				TGetQuantityDiscounts_Return PriceListAssignment_out = PriceListAssignments_out[i];

				if (PriceListAssignment_out.Error == null) 
				{
					Console.WriteLine("successfully retrieved QuantityDiscounts for Product: " + PriceListAssignment_out.Product);
					result.Add(PriceListAssignment_out);
				} 
				else 
				{
					Console.WriteLine("an error occured (Epages Error):\n" + PriceListAssignment_out.Error.Message);
				}
			}

			return result;
		}

		/// <summary>
		/// use stub to create a ValueDiscounts via web services
		/// </summary>
		/// <param name="PriceListAssignments">Array of TSetValueDiscounts_Input</param>
		/// <returns>ArrayList of TSetValueDiscounts_Return</returns>
		public ArrayList setValueDiscounts(TSetValueDiscounts_Input[] PriceListAssignments) 
		{
			TSetValueDiscounts_Return[] PriceListAssignments_out = stub.setValueDiscounts(PriceListAssignments);

			ArrayList result = new ArrayList();

			for(int i = 0; i < PriceListAssignments_out.Length; i++)
			{
				TSetValueDiscounts_Return PriceListAssignment_out = PriceListAssignments_out[i];

				if (PriceListAssignment_out.Error == null) 
				{
					Console.WriteLine("successfully created ValueDiscounts for Product: " + PriceListAssignment_out.Product);
					result.Add(PriceListAssignment_out);
				} 
				else 
				{
					Console.WriteLine("an error occured (Epages Error):\n" + PriceListAssignment_out.Error.Message);
				}
			}

			return result;
		}

		/// <summary>
		/// use stub to get info of a ValueDiscounts via web services
		/// </summary>
		/// <param name="products">Array of product paths</param>
		/// <param name="pricelists">Array of price list paths</param>
		/// <returns>ArrayList of TGetValueDiscounts_Return</returns>
		public ArrayList getValueDiscounts(string[] products, string[] pricelists) 
		{
			TGetValueDiscounts_Return[] PriceListAssignments_out = stub.getValueDiscounts(products, pricelists);

			ArrayList result = new ArrayList();

			for(int i = 0; i < PriceListAssignments_out.Length; i++)
			{
				TGetValueDiscounts_Return PriceListAssignment_out = PriceListAssignments_out[i];

				if (PriceListAssignment_out.Error == null) 
				{
					Console.WriteLine("successfully retrieved ValueDiscounts for Product: " + PriceListAssignment_out.Product);
					result.Add(PriceListAssignment_out);
				} 
				else 
				{
					Console.WriteLine("an error occured (Epages Error):\n" + PriceListAssignment_out.Error.Message);
				}
			}

			return result;
		}

		/// <summary>
		/// use stub to create a ScalePrices via web services
		/// </summary>
		/// <param name="PriceListAssignments">Array of TSetScalePrices_Input</param>
		/// <returns>ArrayList of TSetScalePrices_Return</returns>
		public ArrayList setScalePrices(TSetScalePrices_Input[] PriceListAssignments) 
		{
			TSetScalePrices_Return[] PriceListAssignments_out = stub.setScalePrices(PriceListAssignments);

			ArrayList result = new ArrayList();

			for(int i = 0; i < PriceListAssignments_out.Length; i++)
			{
				TSetScalePrices_Return PriceListAssignment_out = PriceListAssignments_out[i];

				if (PriceListAssignment_out.Error == null) 
				{
					Console.WriteLine("successfully created ScalePrices for Product: " + PriceListAssignment_out.Product);
					result.Add(PriceListAssignment_out);
				} 
				else 
				{
					Console.WriteLine("an error occured (Epages Error):\n" + PriceListAssignment_out.Error.Message);
				}
			}

			return result;
		}

		/// <summary>
		/// use stub to get info of a ScalePrices via web services
		/// </summary>
		/// <param name="products">Array of product paths</param>
		/// <param name="pricelists">Array of price list paths</param>
		/// <returns>ArrayList of TGetValueDiscounts_Return</returns>
		public ArrayList getScalePrices(string[] products, string[] pricelists) 
		{
			TGetScalePrices_Return[] PriceListAssignments_out = stub.getScalePrices(products, pricelists);

			ArrayList result = new ArrayList();

			for(int i = 0; i < PriceListAssignments_out.Length; i++)
			{
				TGetScalePrices_Return PriceListAssignment_out = PriceListAssignments_out[i];

				if (PriceListAssignment_out.Error == null) 
				{
					Console.WriteLine("successfully retrieved ScalePrices for Product: " + PriceListAssignment_out.Product);
					result.Add(PriceListAssignment_out);
				} 
				else 
				{
					Console.WriteLine("an error occured (Epages Error):\n" + PriceListAssignment_out.Error.Message);
				}
			}

			return result;
		}
	}
}
